lib/Smr/Container/DiContainer.php

```diff
@@ -0,0 +1,105 @@
+<?php declare(strict_types=1);
+
+namespace Smr\Container;
+
+use DI\Container;
+use DI\ContainerBuilder;
+use Dotenv\Dotenv;
+use MySqlDatabase;
+use mysqli;
+use Smr\MySqlProperties;
+use function DI\autowire;
+
+/**
+ * Class DiContainer
+ * A wrapper around the DI\Container functionality that will allow
+ * static usage of container methods.
+ * @package Smr\Container
+ */
+class DiContainer {
+	private static DiContainer $instance;
+	private Container $container;
+
+	private function __construct() {
+		$this->container = $this->buildContainer();
+	}
+
+	private function getDefinitions(): array {
+		return [
+			/*
+			 * mysqli is a 3rd party library, and we do not have control over its constructor.
+			 * In order for PHP-DI to construct an instance of a class, each constructor argument must
+			 * be able to be constructed.
+			 * PHP-DI cannot construct mysqli by itself, because all of its arguments are primitive types.
+			 * Therefore, we need to declare a provider factory for the container to use when constructing new instances.
+			 *
+			 * The factories themselves are able to use dependency injection as well, so we can provide the MysqlProperties
+			 * typehint to make sure the container constructs and instance and provides it to the factory.
```

Comment:
```suggestion
			 * The factories themselves are able to use dependency injection as well, so we can provide the MySqlProperties
			 * typehint to make sure the container constructs an instance and provides it to the factory.
```
Just fixing some typos.